### PR TITLE
Initialize from file, Add entrainment, add resistance

### LINKED ISSUE #321: Option to read the initial particle distribution from file
I think this warants a new function initParticlesFromFile, right? Since this is something we might want to keep to experiment with....

_Originally posted by @fso42 in https://github.com/avaframe/AvaFrame/pull/320#discussion_r580156363_

### PR DESCRIPTION
**Pull request by matthiasto**
Not sure if I added the entrainement and resistance properly.
The naming of the outputs of com1DFAPy is not good (no difference between null and entres simulations...
fix #321

### CODE REVIEW

**Review by fso42 — CHANGES_REQUESTED** [review]
@Matthias: please remove the modName option from the cleanSingleAvaDir. I'm writing an extra function that works similar to the cfg reading

File: avaframe/com1DFAPy/DFAfunctionsCython.pyx [hunk 1]
```diff
@@ -187,12 +193,15 @@ def computeForceC(cfg, particles, fields, dem, Ment, Cres, dT):
   cdef double[:, :] VX = fields['Vx']
   cdef double[:, :] VY = fields['Vy']
   cdef double[:, :] VZ = fields['Vz']
+  cdef double[:, :] Ment = fields['Ment']
+  cdef double[:, :] Cres = fields['Cres']
   cdef long[:] IndCellX = particles['indX']
   cdef long[:] IndCellY = particles['indY']
   cdef long indCellX, indCellY
-  cdef double A, uMag, m, h, x, y, z, ux, uy, uz, nx, ny, nz
+  cdef double A, Aent, cres, uMag, m, dm, h, ment, dEnt, dis
```
Comment (fso42) [inline avaframe/com1DFAPy/DFAfunctionsCython.pyx:201]:
```suggestion
  cdef double A, aEnt, cres, uMag, m, dm, h, mEnt, dEnt, dis
```
Please stick to the variable naming

File: avaframe/com1DFAPy/com1DFACfg.ini [hunk 2]
```diff
@@ -67,9 +67,13 @@ R = 0.05
 B = 4.13
 
 #++++++++++++ Resistance force parameters
-hRes = 1
+hRes = 10
+cw = 0.5
+dRes = 0.3
+sres = 5
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFACfg.ini:73]: Something we need to discuss: maybe have a few presets available (this here is wood) 
Also see general comment -> more descriptive

File: avaframe/com1DFAPy/runCom1DFA.py [hunk 3]
```diff
@@ -50,59 +35,35 @@ def runCom1DFAPy(avaDir='', cfgFile='', relTh='', flagAnalysis=True):
     modName = 'com1DFAPy'
 
     # Clean input directory(ies) of old work and output files
-    initProj.cleanSingleAvaDir(avalancheDir, keep=logName, deleteOutput=False)
+    initProj.cleanSingleAvaDir(avalancheDir, keep=logName, deleteOutput=False, onlyModule=modName)
```
Comment (fso42) [inline avaframe/com1DFAPy/runCom1DFA.py:38]: see general comment -> remove the onlyModule option

File: avaframe/in3Utils/initializeProject.py [hunk 4]
```diff
@@ -46,7 +46,11 @@ def cleanSingleAvaDir(avaDir, keep=None, deleteOutput=True):
     # Try to remove OUTPUTS folder, only pass FileNotFoundError, i.e. folder
     # does not exist
     if deleteOutput:
-        _checkForFolderAndDelete(avaDir, 'Outputs')
+        if onlyModule:
+            fPath = os.path.join(avaDir, 'Outputs')
+            _checkForFolderAndDelete(fPath, onlyModule)
+        else:
+            _checkForFolderAndDelete(avaDir, 'Outputs')
```
Comment (fso42) [inline avaframe/in3Utils/initializeProject.py:53]: see general comment

**Review by fso42 — CHANGES_REQUESTED** [review]
Please make the variables in the cython functions more descriptive. There is no reason to not use longer variable names in cython code. something obvious like uX is fine...

If the list of variables gets too long then,-> split the function

File: avaframe/com1DFAPy/com1DFA.py [hunk 5]
```diff
@@ -112,7 +116,78 @@ def initializeMesh(dem, num=4):
     return dem
 
 
-def initializeSimulation(cfg, relRaster, dem):
+def setDEMorigin(demOri):
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:119]:
```suggestion
def setDEMoriginToZero(demOri):
```

File: avaframe/com1DFAPy/com1DFA.py [hunk 6]
```diff
@@ -375,7 +448,7 @@ def placeParticles(mass, indx, indy, csz, massPerPart):
     return xpart, ypart, mPart, nPart
 
 
-def intializeMassEnt(dem):
+def intializeMassEnt(dem, flagEntRes, entLine):
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:451]:
```suggestion
def initializeMassEnt(dem, flagEntRes, entLine):
```
typo...  -> Maybe even rename it to initializeEntrainmentMass ?

File: avaframe/com1DFAPy/com1DFA.py [hunk 7]
```diff
@@ -112,7 +116,78 @@ def initializeMesh(dem, num=4):
...
+
+    # initialize entrainment and resistance
+    rhoEnt = cfgGen.getfloat('rhoEnt')
+    hEnt = cfgGen.getfloat('hEnt')
+    # flagEntRes = False
+    Ment = intializeMassEnt(demOri, flagEntRes, entLine)
+    Cres = intializeResistance(cfgGen, demOri, flagEntRes, resLine)
+    # surfacic entrainment mass available (unit kg/m²)
+    fields['Ment'] = Ment*rhoEnt*hEnt
+    print(np.sum(fields['Ment']*dem['Area']))
```
Comment (awirb) [inline avaframe/com1DFAPy/com1DFA.py:184]: turn into log message?

**Review by fso42 — CHANGES_REQUESTED** [review]

**Review by awirb — CHANGES_REQUESTED** [review]

File: avaframe/com1DFAPy/DFAfunctionsCython.pyx [hunk 8]
```diff
@@ -283,11 +292,42 @@ def computeForceC(cfg, particles, fields, dem, Ment, Cres, dT):
           elif frictType == 2:
             # coulomb friction type (bottom shear stress)
             tau = mu * sigmaB
+          else:
```
[the remark this answers is not in the document]
Comment (matthiasto) [reply avaframe/com1DFAPy/DFAfunctionsCython.pyx:293]: adding some python to the cython will slow it considerably...

File: avaframe/com1DFAPy/DFAfunctionsCython.pyx [hunk 9]
```diff
@@ -187,12 +190,15 @@ def computeForceC(cfg, particles, fields, dem, Ment, Cres, dT):
   cdef double[:, :] VX = fields['Vx']
   cdef double[:, :] VY = fields['Vy']
   cdef double[:, :] VZ = fields['Vz']
+  cdef double[:, :] MENT = fields['Ment']
+  cdef double[:, :] CRES = fields['Cres']
   cdef long[:] IndCellX = particles['indX']
   cdef long[:] IndCellY = particles['indY']
   cdef long indCellX, indCellY
-  cdef double A, uMag, m, h, x, y, z, ux, uy, uz, nx, ny, nz
+  cdef double A, aCell, aEnt, cRes, uMag, m, dm, h, mEnt, dEnt, dis
```
[the remark this answers is not in the document]
Comment (fso42) [reply avaframe/com1DFAPy/DFAfunctionsCython.pyx:198]: That's what I meant with more descriptive (therefore easier to read) names in my general comment. 
i.e. areaCell or cellArea  or entrMass/entrainmentMass/cellEntrainmentMass/cellEntrMass etc make it a lot easier to understand

**Pull request merged**